### PR TITLE
Added SCC algo

### PR DESCRIPTION
**Pull request by ramalik00**
Added SCC  Algorithm

### CODE REVIEW

File: c++/ssc_kosaraju_algo.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
+//Somos insignificantes. Por mais que você programe sua vida, a qualquer momento tudo pode mudar.
```
Comment (n4i9kita) [inline c++/ssc_kosaraju_algo.cpp:1]: #1 , just curious, why would someone write a quote  in Portuguese here ? 👀

Comment (ramalik00) [reply c++/ssc_kosaraju_algo.cpp:1]: Sorry that was part of my template although it serves no purpose in implementation .
It's just that I like this quote and used in my template and forgot to edit it out

Comment (n4i9kita) [reply c++/ssc_kosaraju_algo.cpp]: also, you were supposed to push it from a branch other than main.

Comment (ramalik00) [reply c++/ssc_kosaraju_algo.cpp]: Sure 
I'll take care of it from next time 👍

File: c++/ssc_kosaraju_algo.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
+
+// If you have God on your side,everything becomes clear
+
+#include <bits/stdc++.h>
+#include <ext/pb_ds/assoc_container.hpp>
+#include <ext/pb_ds/tree_policy.hpp>
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:6]: not useful for algorithm and also use only required libraries

File: c++/ssc_kosaraju_algo.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+using namespace std;
+using namespace __gnu_pbds;
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:9]: again not needed from algo's point of view

File: c++/ssc_kosaraju_algo.cpp [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    while (b > 0) {
+        if (b & 1)
+            res = res * a % MOD;
+        a = a * a % MOD;
+        b >>= 1;
+    }
+    res=res%MOD;
+    return res;
+}
+const int MAXN=100001;
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:61]: same as above

File: c++/ssc_kosaraju_algo.cpp [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+vector<vector<int>> adj, adj_rev;
+vector<bool> used;
+vector<int> order, component;
+int n;
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:65]: dont use global variables

File: c++/ssc_kosaraju_algo.cpp [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+void dfs1(int v) {
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:66]: SCC can be found using a single dfs function you can just add another parameter to check whether to do dfs of type one or two.

File: c++/ssc_kosaraju_algo.cpp [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        cin>>u>>v;
+        adj[u].push_back(v);
+        adj_rev[v].push_back(u);
+    }
+    used.assign(n+1, false);
+
+    for (int i = 1; i <= n; i++)
+        if (!used[i])dfs1(i);
+
+    reverse(order.begin(), order.end());
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:105]: this purpose can be served better by a stack

File: c++/ssc_kosaraju_algo.cpp [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    used.assign(n+1, false);
+    int cnt=1;
+    for (auto v : order){
+        if (!used[v]) {dfs2 (v);
+        //Print Component
+        cout<<"Component : "<<cnt<<endl;
+        for(auto&j:component)cout<<j<<" ";
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:112]: use a `list` or `queue` for the same for better time complexity

File: c++/ssc_kosaraju_algo.cpp [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+int main() {
+    ios_base::sync_with_stdio(0);
+    cin.tie(0); cout.tie(0);
+    
+    #ifdef LOCAL
+        freopen("input.txt", "r", stdin);
+        freopen("output.txt", "w", stdout);
+    #endif
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:129]: no needed

File: c++/ssc_kosaraju_algo.cpp [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    
+    int tc=1;// cin >> tc;
+    for (int t = 1; t <= tc; t++) {
+        //cout << "Case #" << t  << ": ";
+        solve();
+    }
```
Comment (king-11) [inline c++/ssc_kosaraju_algo.cpp:135]: just have to write the algo no test cases not a contest so just write the solve function here

**Review by king-11 — CHANGES_REQUESTED** [review]

**Comment by king-11** [comment]
Except for the changes in algorithm code one thing that you had to take care of was to create a new branch for PR and not use `main` so please do that and with updated changes make a new PR

**Comment by king-11** [comment]
Close for #15

**Pull request closed without merging**